src/compiler/parser-expr-variable.bas

```diff
@@ -433,7 +434,7 @@ sub cUdtTypeMember _
 			'' ID
 			lexSkipToken( LEXCHECK_POST_SUFFIX )
 			dtype = symbGetFullType( sym )
-			subtype = symbGetSubType( sym )
+			subtype = sym
```

Comment:
This doesn't seem right; it would only be correct if sym is always a struct, but that's not guaranteed here, or is it?

Reply:
Originally, with `subtype = symbGetSubType( sym )` I had this issue:
```
type T1
        __ as integer
        type T2
                N as integer
        end type
end type
print sizeof( T1.T2.N )

error 265: Symbol not a CLASS, ENUM, TYPE or UNION type, before 'N' in 'print sizeof( T1.T2.N )'
```
I thought what is happening here is that `cUdtTypeMember()` is only called after a failed attempt at `cTypeOrExpression()` which still passes back a dtype+subtype (struct+T1.T2).  Since the subtype is used as the parent in the call to `hMemberId()`, then getting the subtype of the subtype was skipping over a type level.  Checking for non-struct is at the top of the loop.

Reply:
Ah, I think I see the issue now: The dtype/subtype need to be updated as new context for the next iteration of that while loop, just as in `cUdtMember()`, to parse multiple "nesting" levels of `.` member accesses. I didn't know this was possible nowadays without using `dim`/`var` to define variables/instances, that's really nice.

Anyways, I've been trying to break it:
```
type T1
	innerfield as short
end type

type T2
	myfield as T1
end type

print sizeof( T2.myfield.innerfield )
```
result:
```
src/compiler/symb.bas(1429): assertion failed at SYMBLOOKUPAT: (ns->class =FB_SYMBCLASS_STRUCT) or (ns->class =FB_SYMBCLASS_NAMESPACE) or (ns->class =FB_SYMBCLASS_ENUM)
```
Looks to me like both situations are possible here - finding fields, methods, static variables as usual, but also the nested types now. So I think this code needs to check the symbol class in order to know how to update dtype/subtype. Basically treating types (structs) as a special case, and keep the old version for other cases.

Reply:
Thanks, your insight helps.  Agreed, existing code paths should stay the same.

I think I will also need to put something in `cUdtMember()` to error when we expect an expression but access the nested type (declaration) instead of a field:
~~~
type outer
	a as byte
	type inner
		a as short
	end type
end type

dim x as outer
print sizeof( x.inner.a )
~~~
I started adding tests to the test suite which will be included in the next commits.

Reply:
Reverted to previous logic and added separate case in [c93f26676ba9f9a5aab7f7028476cb53de1a8512]

This should address the issues of the last couple of posts in this conversation.